integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java

```diff
@@ -193,17 +210,72 @@ public static AlluxioConfiguration tryLoadingConfigFromMaster(FileSystemContext
     return fsContext.getClusterConf();
   }
 
+  /**
+   * Fills the path status.
+   *
+   * @param policy auth policy
+   * @param stat file stat to fill
+   * @param status status
+   */
+  public static void fillStat(AuthPolicy policy, FileStat stat, URIStatus status) {
+    updateStatSize(stat, status.getLength());
+
+    final long ctime_sec = status.getLastModificationTimeMs() / 1000;
+    final long atime_sec = status.getLastAccessTimeMs() / 1000;
+    // Keeps only the "residual" nanoseconds not caputred in citme_sec
+    final long ctime_nsec = (status.getLastModificationTimeMs() % 1000) * 1_000_000L;
+    final long atime_nsec = (status.getLastAccessTimeMs() % 1000) * 1_000_000L;
+
+    stat.st_atim.tv_sec.set(atime_sec);
+    stat.st_atim.tv_nsec.set(atime_nsec);
+    stat.st_ctim.tv_sec.set(ctime_sec);
+    stat.st_ctim.tv_nsec.set(ctime_nsec);
+    stat.st_mtim.tv_sec.set(ctime_sec);
+    stat.st_mtim.tv_nsec.set(ctime_nsec);
+
+    stat.st_uid.set(policy.getUid(status.getOwner())
+        .orElse(AlluxioFuseUtils.ID_NOT_SET_VALUE));
+    stat.st_gid.set(policy.getGid(status.getGroup())
+        .orElse(AlluxioFuseUtils.ID_NOT_SET_VALUE));
+
+    int mode = status.getMode();
+    if (status.isFolder()) {
+      mode |= FileStat.S_IFDIR;
+    } else {
+      mode |= FileStat.S_IFREG;
+    }
+    stat.st_mode.set(mode);
+    stat.st_nlink.set(1);
+  }
+
+  /**
+   * Updates file status size.
+   *
+   * @param stat stat to file
+   * @param size size
+   */
+  public static void updateStatSize(FileStat stat, long size) {
+    stat.st_size.set(size);
+
+    // Sets block number to fulfill du command needs
+    // `st_blksize` is ignored in `getattr` according to
+    // https://github.com/libfuse/libfuse/blob/d4a7ba44b022e3b63fc215374d87ed9e930d9974/include/fuse.h#L302
+    // According to http://man7.org/linux/man-pages/man2/stat.2.html,
+    // `st_blocks` is the number of 512B blocks allocated
+    stat.st_blocks.set((int) Math.ceil((double) size / 512));
+  }
+
   /**
    * @return the system uid
    */
   public static long getSystemUid() {
     String launchUser = System.getProperty("user.name");
     if (launchUser == null || launchUser.isEmpty()) {
-      throw new RuntimeException("Failed to get current system user name");
+      throw new UnavailableRuntimeException("Failed to get current system user name");
```

Comment:
Is this user mistake or alluxio internal mistake?

Reply:
it may be a system problem, not alluxio internal but also not user mistake. FailedPreconsiditon?

Reply:
it can be Alluxio internal error (not write the code to be able to get the user.name) or external error (operating system does not provide that info).
probably UnavailableRuntimeException suits better, but normally this one is not retryable but our unavailableRuntimeException is retryable.
Could not find another better runtime exception

Reply:
Would prefer to say it's a user error since System.getProperty should return the correct result if the user sets the system correctly. We can make unavailable exception not retryable for this specific case if you think it's a better choice. But my personal feeling(although I don't get many contexts) is that Failedprecondition would be a better fit since it means that the client should not retry until the system state(user name) has been explicitly fixed.